client/src/TenderClient.js

```diff
@@ -0,0 +1,78 @@
+import axios from "axios";
+
+class TenderClient {
+	constructor(baseURL, token) {
+		this.client = axios.create({
+			baseURL: baseURL,
+			headers: {
+				Authorization: `Bearer ${token}`,
+				"Content-Type": "application/json",
+			},
+		});
+
+		this.client.interceptors.response.use(
+			(response) => response,
+			(error) => {
+				console.error(
+					"API call error:",
+					error.response ? error.response.data : error.message
+				);
+				return Promise.reject(error);
+			}
+		);
+	}
+
+	async get(endpoint) {
+		try {
+			const response = await this.client.get(endpoint);
+			return response.data;
+		} catch (error) {
```

Comment:
If the error is a `401`, delete the token from local storage and redirect the user to the log-in page. Otherwise, re-throw the error.

Do this for all methods.